scripts/tofu/dev-provider.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# ./setup-provider.sh <provider> <version>
+
+set -e 
+
+# Validate user input
+if [ $# -ne 2 ]; then
+  echo "Usage: $0 <provider> <giturl>"
+  exit 1
+fi
+
+PROVIDER=$1
+GITURL=$2
+VERSION="0.0.0-dev"
+
+OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+ARCH=$(uname -m)
+
+if [ "$OS" = "darwin" ] && [ "$ARCH" = "arm64" ]; then
+  PLATFORM="darwin_arm64"
+else
+  PLATFORM="linux_amd64"
+fi
+
+# Download binary
+DIR=~/.terraform.d/plugins/terraform.local/${PROVIDER}/${PROVIDER}/${VERSION}/${PLATFORM}
+mkdir -p $DIR
+(umask u=rwx,g=rwx,o=rwx && mkdir -p $DIR)
+git clone $GITURL provider-clone
```

Comment:
Not a huge deal, but it would be a good idea to remove this cloned directory at the end of this script.
Take a look at the `trap` function in bash.
You can do something like `trap 'rm -rf provider-clone' EXIT` after the `set -e` to remove the cloned directory after this script runs.